modules/test/src/main/java/io/oasp/module/test/common/base/ComponentTest.java

```diff
@@ -22,6 +22,57 @@
 @RunWith(SpringJUnit4ClassRunner.class)
 @TestExecutionListeners({ TransactionalTestExecutionListener.class, DependencyInjectionTestExecutionListener.class })
 @Category(CategoryComponentTest.class)
-public abstract class ComponentTest extends Assertions {
+public abstract class ComponentTest extends BaseTest {
+
+  protected DbTestHelper dbTestHelper;
+
+  private static boolean DB_NEEDS_RESET = true;
+
+  @Override
+  protected void doSetUp() {
+
+    doDatabaseSetUp();
+  }
+
+  @Override
+  protected void doTearDown() {
+
+    doDatabaseTearDown();
+  }
+
+  protected void doDatabaseSetUp() {
+
+    if (dbNeedsReset()) {
```

Comment:
I guess we have a missunderstanding here.
The sub-class should have the possibility to override a boolean method ```isResetDb()``` in order to decide if the DB should be reset only once per test or for each test-method. If the test is readonly this can save performance/time.